### PR TITLE
Adopt copier

### PR DESCRIPTION
**Pull request by DiamondJoseph**
Adopts the Copier template at 2.1.0, which deprecates the previous Skeleton projects from Diamond.

### CODE REVIEW

File: Dockerfile
Comment (callumforrester) [inline Dockerfile:1]: Does this need a `Dockerfile`? Could change to "no" in the copier answers

**Review by callumforrester — APPROVED** [review]

File: .vscode/extensions.json [hunk 1]
```diff
@@ -1,10 +1,5 @@
 {
     "recommendations": [
         "ms-vscode-remote.remote-containers",
-        "ms-python.python",
-        "tamasfe.even-better-toml",
-        "redhat.vscode-yaml",
-        "ryanluker.vscode-coverage-gutters",
-        "charliermarsh.Ruff"
     ]
```
Comment (stan-dot) [inline .vscode/extensions.json:4]: I think we should keep toml, yaml and python and ruff extensions

File: catalog-info.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: i22-bluesky
+  title: i22-bluesky
+  description: Plans and behaviours specific to the i22 beamline at DiamondLightSource.
+spec:
+  type: documentation
+  lifecycle: experimental
+  owner: user/xma12127
```
Comment (stan-dot) [inline catalog-info.yaml:10]: owner should be a group, not an individual

File: pyproject.toml [hunk 3]
```diff
@@ -46,8 +41,8 @@ i22-bluesky = "i22_bluesky.__main__:main"
 GitHub = "https://github.com/DiamondLightSource/i22-bluesky"
 
 [[project.authors]] # Further authors may be added by duplicating this section
-email = "joseph.ware@diamond.ac.uk"
-name = "Ware, Joseph (DLSLtd,RAL,LSCI)"
+email = "stanislaw.malinowski@diamond.ac.uk"
+name = "Malinowski, Stanisław (DLSLtd,RAL,LSCI)"
```
Comment (stan-dot) [inline pyproject.toml:45]: please keep multiple authors

File: src/i22_bluesky/plans/stopflow.py [hunk 4]
```diff
@@ -17,14 +17,13 @@
...
 
 import bluesky.plan_stubs as bps
 import bluesky.plans as bp
 import bluesky.preprocessors as bpp
 from bluesky.protocols import Readable
-from dls_bluesky_core.core import MsgGenerator
-from dodal.common import inject
+from dodal.common import MsgGenerator, inject
 from dodal.devices.tetramm import TetrammDetector
 from dodal.plans.data_session_metadata import attach_data_session_metadata_decorator
```
Comment (stan-dot) [inline src/i22_bluesky/plans/stopflow.py:28]: is this an unrelated dodal change?

**Review by stan-dot — CHANGES_REQUESTED** [review]
some docs I disagree with

File: .vscode/extensions.json (hunk 1 above)
Comment (DiamondJoseph) [reply .vscode/extensions.json:4]: We can re-add at a later point, these came from the skeleton and are not part of the copier template. I prefer a clean starting point from which we can later add extensions again.

File: src/i22_bluesky/plans/stopflow.py (hunk 4 above)
Comment (DiamondJoseph) [reply src/i22_bluesky/plans/stopflow.py:28]: This is a requirement as dls_blueksy_core (which is deprecated) requires ScanSpec, which uses an outdated version of Pydantic without pinning it.

File: catalog-info.yaml (hunk 2 above)
Comment (DiamondJoseph) [reply catalog-info.yaml]: I've transfered ownership to data-acquisition group.

File: pyproject.toml (hunk 3 above)
Comment (DiamondJoseph) [reply pyproject.toml]: I've left myself as an additional owner but put you in the copier answers to imply ownership is yours.

Comment (stan-dot) [reply pyproject.toml]: ok I guess

File: catalog-info.yaml (hunk 2 above)
Comment (stan-dot) [reply catalog-info.yaml]: fantastic!

**Review by stan-dot — APPROVED** [review]
all good now
